PHP Fatal error: Uncaught Phan\Exception\InvalidFQSENException in 1.1.6

Nightly automated CI last night (Saturday night) sucked in the latest ``phan`` 1.1.6 here:
https://github.com/owncloud/data_exporter/
https://drone.owncloud.com/owncloud/data_exporter/270/49
```
+ make test-php-phan
phan composer.lock is not up to date.
composer bin phan install --no-progress
Changed current directory to vendor-bin/phan
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 15 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-mbstring (v1.10.0): Loading from cache
  - Installing symfony/contracts (v1.0.2): Loading from cache
  - Installing symfony/console (v4.2.1): Loading from cache
  - Installing psr/log (1.1.0): Loading from cache
  - Installing sabre/event (5.0.3): Loading from cache
  - Installing microsoft/tolerant-php-parser (v0.0.15): Loading from cache
  - Installing webmozart/assert (1.3.0): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
  - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.3.0): Loading from cache
  - Installing netresearch/jsonmapper (v1.4.0): Loading from cache
  - Installing felixfbecker/advanced-json-rpc (v3.0.3): Loading from cache
  - Installing composer/xdebug-handler (1.3.1): Loading from cache
  - Installing composer/semver (1.4.2): Loading from cache
  - Installing phan/phan (1.1.6): Downloading (100%)
symfony/contracts suggests installing psr/cache (When using the Cache contracts)
symfony/contracts suggests installing psr/container (When using the Service contracts)
symfony/contracts suggests installing symfony/cache-contracts-implementation
symfony/contracts suggests installing symfony/service-contracts-implementation
symfony/contracts suggests installing symfony/translation-contracts-implementation
symfony/console suggests installing psr/log-implementation (For using the console logger)
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing symfony/process
Writing lock file
Generating autoload files
php -d zend.enable_gc=0 vendor-bin/phan/vendor/bin/phan --config-file .phan/config.php --require-config-exists
PHP Fatal error:  Uncaught Phan\Exception\InvalidFQSENException: Invalid namespaced name in /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/Language/FQSEN/FullyQualifiedGlobalStructuralElement.php:152
Stack trace:
#0 /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/Language/FQSEN/FullyQualifiedGlobalStructuralElement.php(217): Phan\Language\FQSEN\FullyQualifiedGlobalStructuralElement::make('\\', 'OCI-Lob', 0)
#1 /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/Memoize.php(81): Phan\Language\FQSEN\FullyQualifiedGlobalStructuralElement::Phan\Language\FQSEN\{closure}()
#2 /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/Language/FQSEN/FullyQualifiedGlobalStructuralElement.php(219): Phan\Language\FQSEN\FullyQualifiedGlobalStructuralElement::memoizeStatic('Phan\\Language\\F...', Object(Closure))
#3 /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/CodeBase.php(708): Phan\Language\FQSE in /var/www/owncloud/apps/data_exporter/vendor-bin/phan/vendor/phan/phan/src/Phan/Language/FQSEN/FullyQualifiedGlobalStructuralElement.php on line 152
Makefile:87: recipe for target 'test-php-phan' failed
make: *** [test-php-phan] Error 255 
```

``InvalidFQSENException`` looks like it is a new problem. This CI passed when it used ``phan`` 1.1.5 on Friday night: https://drone.owncloud.com/owncloud/data_exporter/269/49

The crash is in ``phan`` itself. It is not reporting to me information about something invalid in the code being analyzed???

--- a/src/Phan/CodeBase.php
+++ b/src/Phan/CodeBase.php
@@ -706,9 +706,13 @@ class CodeBase
     public function addReflectionClass(ReflectionClass $class)
     {
         // Map the FQSEN to the class
-        // @phan-suppress-next-line PhanThrowTypeAbsentForCall
-        $class_fqsen = FullyQualifiedClassName::fromFullyQualifiedString($class->getName());
-        $this->fqsen_class_map_reflection->offsetSet($class_fqsen, $class);
+        try {
+            $class_fqsen = FullyQualifiedClassName::fromFullyQualifiedString($class->getName());
+            $this->fqsen_class_map_reflection->offsetSet($class_fqsen, $class);
+        } catch (FQSENException $_) {
+            // Fixes uncaught Phan\Exception\InvalidFQSENException for #2222
+            // Just give up on analyzing uses of the class "OCI-Lob" and anything similar - It's invalid because of the hyphen.
+        }
     }
 
     /**

